lib/web/websocket/connection.js

```diff
@@ -308,7 +308,7 @@ function onSocketClose () {
   if (result) {
     code = result.code ?? 1005
     reason = result.reason
-  } else if (isClosed(ws) && !ws[kReceivedClose]) {
```

Comment:
onSocketClose is called once the socket is closed. By the RFC that means we are in the CLOSED state already, we just haven't set it yet.